core/aws-core/src/main/java/software/amazon/awssdk/awscore/AwsServiceClientConfiguration.java

```diff
@@ -112,6 +101,11 @@ public final ClientOverrideConfiguration overrideConfiguration() {
         public final Region region() {
             return region;
         }
+
+        @Override
+        public final URI endpointOverride() {
```

Comment:
Shouldn't we have a setter on the interface to override the return type?

Reply:
added overridden setters for endpointOverride and overrideConfiguration